### ISSUE TITLE
Error when adding buttons to the footer of a Modal subclass

### ISSUE THREAD

**Issue by jchappelle** [l0rdn1kk0n/wicket-bootstrap#713]
I'm loving this library. Really makes it easy to modernize a wicket app. Thank you for writing it.

However, I'm having an issue with getting a modal window to work. It's really only when adding buttons to the footer via calling the addButton method.

It looks strange to me that the Modal.html file is using a wicket:fragment and then binding it to a ListView in the code. I'm not an expert at fragments so I'm not sure if that's an issue.

As a work around I'm going to move on by adding my buttons to my content panel instead until I hear back from you. 

Thank you for your time.

Here's the error:
```
1. [ListView [Component id = buttons, page = com.redi.verifi.client.wicket.pages.activity.history.HistoryPage, path = bodyPanel:addEditModal:dialog:footer:buttons, type = org.apache.wicket.markup.html.list.ListView, isVisible = true, isVersioned = true, markup = [markup = file:/C:/eclipse-workspaces/redi/verifi-yh-parent-TRUNK/verifi-yh-wicket/target/classes/com/redi/verifi/client/wicket/pages/activity/history/AddEditTransactionFilterPanel.html
<wicket:fragment wicket:id="buttons">
<button type="button" wicket:id="button"></button>
</wicket:fragment>, index = 0, current =  '<wicket:fragment wicket:id="buttons">' (line 0, column 0)]], children =  [ListItem [Component id = 0]]]
2. [ListItem [Component id = 0, page = com.redi.verifi.client.wicket.pages.activity.history.HistoryPage, path = bodyPanel:addEditModal:dialog:footer:buttons:0, type = org.apache.wicket.markup.html.list.ListItem, isVisible = true, isVersioned = true, markup = [markup = file:/C:/eclipse-workspaces/redi/verifi-yh-parent-TRUNK/verifi-yh-wicket/target/classes/com/redi/verifi/client/wicket/pages/activity/history/AddEditTransactionFilterPanel.html
<wicket:fragment wicket:id="buttons">
<button type="button" wicket:id="button"></button>
</wicket:fragment>, index = 0, current =  '<wicket:fragment wicket:id="buttons">' (line 0, column 0)]], children =  [AjaxLink [Component id = button]]]
3. [AjaxLink [Component id = button, page = com.redi.verifi.client.wicket.pages.activity.history.HistoryPage, path = bodyPanel:addEditModal:dialog:footer:buttons:0:button, type = org.apache.wicket.ajax.markup.html.AjaxLink, isVisible = true, isVersioned = true, markup = [markup = file:/C:/eclipse-workspaces/redi/verifi-yh-parent-TRUNK/verifi-yh-wicket/target/classes/com/redi/verifi/client/wicket/pages/activity/history/AddEditTransactionFilterPanel.html
<button type="button" wicket:id="button"></button>, index = 0, current =  '<button type="button" wicket:id="button">' (line 0, column 0)]]]

	at org.apache.wicket.Page.checkRendering(Page.java:666) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.Page.onAfterRender(Page.java:821) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.markup.html.WebPage.onAfterRender(WebPage.java:224) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.Component.afterRender(Component.java:919) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.Component.render(Component.java:2341) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.Page.renderPage(Page.java:1018) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.request.handler.render.WebPageRenderer.renderPage(WebPageRenderer.java:124) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.request.handler.render.WebPageRenderer.respond(WebPageRenderer.java:236) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.core.request.handler.RenderPageRequestHandler.respond(RenderPageRequestHandler.java:175) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.request.cycle.RequestCycle$HandlerExecutor.respond(RequestCycle.java:895) ~[wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.request.RequestHandlerStack.execute(RequestHandlerStack.java:64) ~[wicket-request-7.1.0.jar:7.1.0]
	at org.apache.wicket.request.cycle.RequestCycle.execute(RequestCycle.java:265) [wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.request.cycle.RequestCycle.processRequest(RequestCycle.java:222) [wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.request.cycle.RequestCycle.processRequestAndDetach(RequestCycle.java:293) [wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.protocol.http.WicketFilter.processRequestCycle(WicketFilter.java:261) [wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.protocol.http.WicketFilter.processRequest(WicketFilter.java:203) [wicket-core-7.9.0.jar:7.9.0]
	at org.apache.wicket.protocol.http.WicketFilter.doFilter(WicketFilter.java:284) [wicket-core-7.9.0.jar:7.9.0]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668) [jetty-servlet-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.springframework.orm.hibernate4.support.OpenSessionInViewFilter.doFilterInternal(OpenSessionInViewFilter.java:151) [spring-orm-4.2.6.RELEASE.jar:4.2.6.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.2.6.RELEASE.jar:4.2.6.RELEASE]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668) [jetty-servlet-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:225) [websocket-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668) [jetty-servlet-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:581) [jetty-servlet-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548) [jetty-security-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1158) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511) [jetty-servlet-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1090) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:213) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:109) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:119) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.Server.handle(Server.java:517) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:306) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:242) [jetty-server-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:261) [jetty-io-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95) [jetty-io-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:75) [jetty-io-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceAndRun(ExecuteProduceConsume.java:213) [jetty-util-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:147) [jetty-util-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:654) [jetty-util-9.3.5.v20151012.jar:9.3.5.v20151012]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:572) [jetty-util-9.3.5.v20151012.jar:9.3.5.v20151012]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_51]

```
Here's my java class
```
import org.apache.wicket.ajax.AjaxRequestTarget;
import org.apache.wicket.ajax.markup.html.AjaxLink;
import org.apache.wicket.markup.html.form.*;
import org.apache.wicket.model.*;
import org.apache.wicket.spring.injection.annot.SpringBean;

import com.redi.verifi.client.TransactionFilter;
import com.redi.verifi.client.transactions.TransactionService;

import de.agilecoders.wicket.core.markup.html.bootstrap.dialog.Modal;

public class AddEditTransactionFilterPanel extends Modal<TransactionFilter>
{
    @SpringBean(name="transactionService")
    private TransactionService transactionService;

    public AddEditTransactionFilterPanel(String id, final IModel<TransactionFilter> model)
    {
        super(id, model);

        setOutputMarkupId(true);
        
        Form<TransactionFilter> form = new Form<>("form", new CompoundPropertyModel<TransactionFilter>(getModel()));
        form.add(new RequiredTextField<String>("title").setLabel(Model.of("Filter name")));
        addButton(new AjaxLink<Void>(Modal.BUTTON_MARKUP_ID)
		{
			@Override
			public void onClick(AjaxRequestTarget target)
			{
				close(target);
			}
		}.setBody(Model.of("Cancel")));
        add(form);

    }
}
```

Here's my html file:
```
<?xml version="1.0" encoding="UTF-8"?>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:wicket="http://wicket.apache.org">
<body>
<wicket:extend>
    <form wicket:id="form">
		<div class="form-group">
			<label>Filter name</label>
			<input type="text" wicket:id="title" class="form-control" placeholder="Filter name"></input>
		</div>
    </form>
</wicket:extend>
</body>
</html>
```

**Comment by jchappelle** [comment]
I'm sure you saw my comment in my pull request. In summary, I was using an old version(0.10.6). I found the newer version has the fix. My pull request has a test to prove Modal renders fine when adding buttons.

**Comment by martin-g** [comment]
> Really makes it easy to modernize a wicket app. Thank you for writing it.

No! All you need to "modernize any web app" is a web designer with good HTML/CSS skills!

Wicket (core) doesn't have any opinion on your look & feel!
Whatever HTML and CSS the application developer applies this is what will be presented to the end user!

And, IMHO, Bootstrap was modern 3 years ago :-)

**Comment by martin-g** [comment]
Please provide a quickstart app (http://wicket.apache.org/start/quickstart.html) so we can debug it!

I agree with the usage of Fragments in some places. Few months ago I've fixed/reworked some of it but it seems there is more ...

**Comment by martin-g** [comment]
As said in #714 - this is no more an issue.

**Comment by jchappelle** [comment]
> No! All you need to "modernize any web app" is a web designer with good HTML/CSS skills!
I agree. But I've used Wicket for 10 years and I came from a Swing background before that. So I don't know that much about UX or what color to paint my bedroom but I do know that bootstrap and your integration has made it super easy to make an enterprise application look like a professional designer has been involved.

The main component I've used is the Navbar. The rest have mostly been styles and the easy install method in the Application.init method. I've replaced all TabbedPanels with BootstrapTabbedPanel. I also decided to start using this because of many other components that are available. I'm migrating a large 5 year old wicket app so I'm eating the elephant one bite at a time and only replacing the pieces that have the most bang for the buck. 

So good work! I hope you keep up the project and I hope to contribute more now that I'm invested in it.
